### ISSUE TITLE
stm32 own build-in watchdog

### ISSUE THREAD

**Issue by rusefi** [rusefi/rusefi#1339]
Are we interested to have stm32 watchdog enabled?

Watchdog impl https://github.com/rusefi/rusefi/pull/5751/
Detect and print Reset cause for STM32 https://github.com/rusefi/rusefi/pull/5783
actually enable watchdog for F4 https://github.com/rusefi/rusefi/pull/5792/

See also https://github.com/rusefi/rusefi/pull/5752
unneeded https://github.com/rusefi/rusefi/pull/5794

**Comment by rusefillc** [comment]
watchdog shall leave a backup RAM bread crumb and reboot if an issue is defected with the firmware
in light of the bread crump we shall software fatal on next boot with "hard fault detected" critical message

**Comment by rusefillc** [comment]
https://github.com/rusefi/rusefi/pull/5792/

**Comment by rusefillc** [comment]
``
set_watchdog_reset 5000
``

**Comment by rusefillc** [comment]
nope, it was not enabled for a year and still is not enabled as of right now @andreika-git
